File: ENGG1420/assignment2/Question4/Question4/src/main/java/Question4.java
//Author//Author: Aiman Kassam (User ID: 1228282)
//Date Created: 2023-01-23
//Assignment #2, Question #4
//Calculator that takes user input for numbers, taking the sum of all the numbers you enter in a loop, breaking when you enter 0 and printing the sum.

//Requried to allow user input.
import java.util.Scanner;
public class Question4 {

    public static void main(String[] args) 
    {
        //Sets variable to be called whenever we want to take user input.
        Scanner scan = new Scanner(System.in);
        
        //Initialize variables to hold total sum that's printed later and user input.
        double sum = 0, num;
        
        //while loop that will run until broken using the if statement embedded in it.
        while(true)
        {
            //Adds every number entered into the total sum until 0 is entered.
            System.out.println("Enter a number to add: ");
            num = scan.nextDouble();
            sum = sum + num;
            if(num == 0)
                break;
        }
        //Prints the total sum.
        System.out.println("The sum of the numbers you entered is: " +sum);
    }
}

File: ENGG1420/a1/Question4.c
//Author: Aiman Kassam (User ID: 1228282)
//Date Created: 2023-01-11
//Assignment #1, Question #4
//Takes a value inputted by the user, returns the sum of all of the input's factors, including itself.

#include <stdio.h>

//Declaration of divisorSum function. See code below main function.
int divisorSum(int num);

void main()
{
    //Variable used to store inputted value.
    int num;

    printf("Enter a number:\n");

    scanf("%d", &num);

    //Function called, value of sum of factors is printed.
    printf("The sum of this numbers factors are: %d", divisorSum(num));
}

int divisorSum(int num)
{
    //Sum variable which is updated for every factor found in upcoming for loop.
    int sum = 0;

    //Checks for all possible factors of inputted number using (input % tested-number), adding along each value proven true.
    for(int i = 1; i <= num; i++)
    {
        if(num % i == 0)
            sum = sum + i;
    }

    return sum;
}